report.md

```diff
@@ -2,9 +2,47 @@
 
 ## Subsystem 1 (MFA)
 
-## Subsystem 2 (Interface for Therapists & Patients)
-Interface will be written in HTML, PHP and SQL to allow database access. Google reCAPTCHA will be included to prevent bots from trying to mass access the system. Each unique user will have their own unique userID and password for login. Therapists and patients will have different login privileges as follow.
+---
+
+Subsystems 2 to 4 will support the following functionalities with the following parameters:
+
+1. Registration
+    1. IC
+    1. Name
+    1. Email
+    1. Phone
+    1. Address
+    1. Age
+    1. Role
+
+1. Log In
+    1. IC
+    1. Password
+    1. Role
+
+A sample POST query looks something like this:
+
+```
+{
+  "query": "registration",
+  "variables": { "ic": "someValue", ... }
+}
+```
+
+The log in system will be protected Google reCAPTCHA to prevent brute force attacks to access into the accounts of administrators.
+
+We will use GraphQL to facilitate communication between the Client & the Server. GraphQL is susceptible to:
+1. SQL injections & XSS (especially if the input field is a custom type, such as JSON). Therefore, we will need to sanitise user input, and we will be using OWASP ZAP to protect our system from XSS.

```

Comment:
elaborate on how OWASP ZAP helps to prevent XSS?

Reply:
I'll remove it. @gngster OWASP ZAP is meant to test whether there are possibilities of XSS, but it doesn't actually protect the system from it, right?